### DISCUSSION TITLE
Custom exchange — mapping result with added property

### DISCUSSION THREAD

**Discussion by svey** [urql-graphql/urql#1497] [Q&A] [answered]
Trying to append a property to the result but I'm not having much luck. Looking at the [Only One Stream Docs](https://formidable.com/open-source/urql/docs/advanced/authoring-exchanges/#only-one-operations-stream) I thought my custom exchange was valid, (referenced for the mapping of resultOps, not the shared/only one stream) — this is my first custom exchange and it looks like I've missed something. Any help is appreciated! 🙏 

**Custom Exchange**
```
export const alertExchange = (dispatch) => ({ forward }) => (
  ops$
) => pipe(
  forward(ops$),
  map(res => ({ ...res, alert: () => dispatch() }))
);
```

**Exchanges**
```
const exchanges = (dispatch) => [
  alertExchange(dispatch),
  dedupExchange,
  errorExchange({ onError }),
  cacheExchange,
  fetchExchange,
];
```

**Comment by kitten** [comment, marked as answer]
Where are you trying to access the custom addition from? The exchange does look correct but it's worth noting that you'll have to guide yourself by the TypeScript types. We don't guarantee arbitrary properties on operations or results to be passed through. Generally speaking, we don't want this to happen, so it's safer to extend the operation on your result by making a copy of it using `makeOperation` and adding your own metadata on using a property on `operation.context`
